### DISCUSSION TITLE
Session not being saved

### DISCUSSION THREAD

**Discussion by anzau** [nextauthjs/next-auth#3196] [Help]
Hi there,

I'm able to login/register using the credentials provider and able to store sessions in the cookies. But for some reason the session isn't stored in the db?

I have JWT enabled to true as I'm using other providers such as Google/Github and they are able to store sessions but not the credentials. I'm also using the prisma adapter as well. Do I have to manually store in the db like I find/create the user?

Thanks

**Comment by stardvst** [comment]
no, you don't have to manually store the user. it should be automatically stored in db upon login.
how are you setting up the db? do you generate the migration? any errors?

**Reply by anzau** [reply]
So with prisma, I generate the migration and have the tables up with no errors. It works fine with the OAuth providers but when I'm using the credentials one, it doesn't give an error but it doesn't save the session associated with the logged in user to the db. I'm able to register and find the user to login. I'm also able to have the session stored in the cookie with the token but not in the db.

**Comment by balazsorban44** [comment]
It's a design limitation

> The Credentials provider can only be used if JSON Web Tokens are enabled for sessions. Users authenticated with the Credentials provider are not persisted in the database.

https://next-auth.js.org/configuration/providers/credentials
